packages/eip-1193-provider/src/__tests__/index.test.ts

```diff
@@ -0,0 +1,495 @@
+      },
+    });
+  },
+};
+
+describe("Test Turnkey EIP-1193 Provider", () => {
+  let turnkeyClient: TurnkeyClient;
+  let eip1193Provider: TurnkeyEIP1193Provider;
+
+  const apiPublicKey = API_PUBLIC_KEY ?? "";
+  const apiPrivateKey = API_PRIVATE_KEY ?? "";
+  const baseUrl = BASE_URL ?? "";
+  const expectedWalletAddress = EXPECTED_WALLET_ACCOUNT_ETH_ADDRESS;
+  const walletId: UUID = WALLET_ID;
+  const organizationId = ORGANIZATION_ID;
+  const foundryChain = {
+    chainName: foundry.name,
+    chainId: numberToHex(foundry.id),
+    rpcUrls: [ANVIL_RPC_URL],
+  };
+
+  beforeAll(() => {
+    // Sets the initial balance
+    // Balance is reset after each test
+    anvil.setBalance();
+  });
+
+  beforeEach(async () => {
+    turnkeyClient = new TurnkeyClient(
+      { baseUrl },
+      new ApiKeyStamper({
+        apiPublicKey,
+        apiPrivateKey,
+      })
+    );
+    eip1193Provider = await createEIP1193Provider({
+      walletId,
+      organizationId,
+      turnkeyClient,
+      chains: [foundryChain],
+    });
+  });
+
+  afterEach(() => {
+    // Reset the nonce and balance of the expected address
+    anvil.setBalance();
+    anvil.resetNonce();
+  });
+
+  const getEIP1193Provider = (
+    chain: AddEthereumChainParameter = foundryChain
+  ) =>
+    createEIP1193Provider({
+      walletId,
+      organizationId,
+      turnkeyClient,
+      chains: [chain],
+    });
+
+  describe("Connectivity Logic", () => {
+    it("should emit connected when successfully connected", async () => {
+      const provider = await getEIP1193Provider();
+      const onConnected = jest.fn();
+      provider.on("connect", (connectionInfo) => {
+        expect(connectionInfo.chainId).toEqual(foundryChain.chainId);
+        onConnected();
+      });
+
+      await provider.request({ method: "eth_blockNumber" });
+
+      expect(onConnected).toHaveBeenCalled();
+    });
+
+    it("should emit disconnected when connectivity is lost", async () => {
+      // Define a chain configuration with a valid RPC URL initially
+      // Performs a deep clone since we need to modify the rpcUrls
+      const chain = JSON.parse(JSON.stringify(foundryChain));
+      // Create an EIP1193 provider instance configured for the specified chain
+      const provider = await getEIP1193Provider(chain);
+
+      // Setup a mock function to track the 'disconnect' event
+      const onDisconnected = jest.fn();
+      provider.on("disconnect", (error) => {
+        expect(error).toBeInstanceOf(ProviderDisconnectedError); // Assert that the error is an instance of ProviderDisconnectedError
+        onDisconnected(); // Record the disconnect event occurrence
+      });
+
+      // Change the RPC URL to an invalid one to simulate connectivity loss
+      chain.rpcUrls[0] = "https://invalid.rpc.url";
+
+      // Attempt to make a request which should fail due to the invalid RPC URL
+      await provider.request({ method: "eth_blockNumber" }).catch(() => {});
+    });
```

Comment:
should we add `expect(onDisconnected).toHaveBeenCalledTimes(1);` to this case as well?

Reply:
yea good call! in the event that the tests runs and `onDisconnected` doesn't get called and the test runs with no assertions